README.md

```diff
@@ -1,44 +1,59 @@
 | [![check-for-update](https://github.com/prisma/e2e-tests/workflows/check-for-update/badge.svg)](https://github.com/prisma/e2e-tests/actions?query=workflow%3Acheck-for-update)                 | -             |
 
-This repository continuously tests Prisma Client on various operating systems, frameworks and platforms.
 
 You can check out the latest test runs by checking the ["test" workflow results](https://github.com/prisma/e2e-tests/actions?query=workflow%3Atest).
 
 ## How it works
 
-### Dependency Updates
+### Projects and Tests
 
-Renovate is enabled for this repository for all dependencies except `prisma` and `@prisma/client`. Our own script handles upgrading prisma-related dependencies since Renovate is too slow for our use case.
+The e2e tests are defined in `.github/workflows/test.yaml` and `.github/workflows/optional-test.yaml`, and the test projects live in folder of form `foo/bar` in this repository. Each `foo` has one or multiple jobs in the GitHub Actions Workflows, and the `bar`s are part of the matrix per job.
 
-### Prisma Updates
+All of the jobs run `.github/scripts/test-project.sh`, which then executes the test project.
 
-If there is a new version, [Prismo](https://github.com/prisma-bot) works tirelessly to commit and push a bump commit, triggering the e2e tests. This is implemented in `.github/workflows/check-for-update.yaml` using a GitHub Action cron job. Since the cron job is limited to run each 5 minutes, we just run each cron job for exactly 5 minutes and check for updates each 10 seconds in each run. This check only runs in the default branch `dev`.
+The test project can install additional dependencies such as CLIs in the optional `prepare.sh`. 
 
-### Branches and npm channels
+The standard entrypoint to set up your projects is `run.sh`. This includes installing dependencies, deploying etc.
+
+Test are then triggered via `test.sh`.
+
+Any clean up or logging work can then be done in `finally.sh` which is executed in all cases, even when the tests fail.
+
+**Note:** You need to use [yarn](https://yarnpkg.com) as it's used for bumping dependencies; i.e. run `yarn install` as a first step in your `run.sh` script.
+
+**Note:** It's important to add `prisma` as a devDependency and `@prisma/client` as a normal dependency in each project's `package.json`.
+
+### Database 
+
+Most tests use a database. The GitHub workflow creates and provides a `DATABASE_URL` to all jobs, and `.github/scripts/test-project.sh` runs `prisma db push` to make sure the project's schema exists on the database.
 
-The default `dev` branch of this repository contains the examples with the development version of Prisma CLI and Prisma Client (`@dev` on npm). These dependencies are kept up to date with a GitHub Action workflow, which updates them every time a new version of Prisma is released.
+Projects that use an external database use a different environment variable name from `DATABASE_URL`.
```

Comment:
It's all external databases actually?

Reply:
Technically yeah, this means databases that are not ours but hosted on a provider. But that is also not super correct, as we have tests like Netlify that use our database servers, but a different connection string. Hmmm...